internal/table/table.go

```diff
@@ -140,44 +146,59 @@ func renderRow(providedColumns []string, colWidths []int) string {
 
 	result := ""
 
-	// Keep rendering lines until there's no column data left to render
-	for len(strings.Join(columns, "")) != 0 {
-		// Iterate over the columns by their line sizes
-		for n, maxLen := range colWidths {
-			// ignore columns that we do not have data for (they have been filled up with the last colValue already)
-			if len(columns) < n+1 {
-				continue
-			}
+	entries := make([][]colorize.Entry, len(columns))
 
-			colValue := []rune(columns[n])
+	widths := make([]int, len(colWidths))
+	copy(widths, colWidths)
+	for n, maxLen := range colWidths {
+		// ignore columns that we do not have data for (they have been filled up with the last colValue already)
+		if len(columns) < n+1 {
+			continue
+		}
 
-			// Detect multi column span
-			if len(colWidths) > n+1 && len(columns) == n+1 {
-				for _, v := range colWidths[n+1:] {
-					maxLen += v
-				}
+		// Detect multi column span
+		if len(colWidths) > n+1 && len(columns) == n+1 {
+			for _, v := range colWidths[n+1:] {
+				maxLen += v
 			}
+		}
 
-			maxLen = maxLen - (padding * 2)
+		maxLen = maxLen - (padding * 2)
+		widths[n] = maxLen
 
-			// How much of the colValue are we using for this line?
-			end := len(colValue)
-			if end > maxLen {
-				end = maxLen
-			}
+		entries[n] = colorize.GetCroppedText(columns[n], maxLen)
+	}
 
-			skipLineBreak := 0
-			if breakpos := runeSliceIndexOf(colValue, linebreakRune); breakpos != -1 && breakpos < end {
-				end = breakpos + 1
-				skipLineBreak = 1
-			}
+	lines := 0
+	for _, entry := range entries {
+		// Each entry represents a column. The longest column determines
+		// the number of rows the table will have.
+		if len(entry) > lines {
+			lines = len(entry)
+		}
+	}
 
-			suffix := strings.Repeat(" ", maxLen-end)
-			result += pad(string(colValue[0:end-skipLineBreak]) + suffix)
+	// Render each row
+	for i := 0; i < lines; i++ {
+		for n, columnEntry := range entries {
+			maxLen := widths[n]
+			text := ""
+			suffix := strings.Repeat(" ", maxLen)
+
+			if len(columnEntry) > i {
+				end := len(columnEntry[i].Line)
```

Comment:
The entry literally has a length property for this purpose. Now you're just including the color codes again.

Reply:
This and the check you commented on above had to be added as we were calculating column widths with the colour tags. I've removed that in the width calculation and now we can use `entry.Length` as we should.